### PR TITLE
Rename `disposeAllDisposables to `dispose` lineup with VSCode

### PR DESCRIPTION
**Pull request by DonJayamanne**
Lineup with VS Cod code, simple rename of a function.

### CODE REVIEW

File: src/platform/common/utils/lifecycle.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { traceWarning } from '../../logging';
```
Comment (rzhao271) [inline src/platform/common/utils/lifecycle.ts:4]: Is this entire file needed, and does the copyright header match the others in this repository?

File: src/platform/common/utils/iterable.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export namespace Iterable {
```
Comment (rzhao271) [inline src/platform/common/utils/iterable.ts:5]: Is this file needed?

File: src/platform/common/utils/functional.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export function once<T extends Function>(this: unknown, fn: T): T {
```
Comment (rzhao271) [inline src/platform/common/utils/functional.ts:4]: Is this file needed?

File: src/platform/common/utils/lifecycle.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/platform/common/utils/lifecycle.ts:4]: Yes for both

File: src/platform/common/utils/iterable.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/platform/common/utils/iterable.ts:5]: Yes

File: src/platform/common/utils/functional.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/platform/common/utils/functional.ts:4]: Yes, if you check this function is used

**Review by rebornix — APPROVED** [review]
